registry/storage/filewriter.go

```diff
@@ -72,7 +72,7 @@ func newFileWriter(ctx context.Context, driver storagedriver.StorageDriver, path
 	buffered := bufferedFileWriter{
```

Comment:
This needs to `&bufferedFileWriter`. That way, the address below will remain valid.